src/main/java/cs361/battleships/models/Board.java

```diff
@@ -3,46 +3,72 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
 public class Board {
+    @JsonProperty private List<Ship> ships; 
+    @JsonProperty private List<Result> attacks;
+
+    /*
 
-	/*
 	DO NOT change the signature of this method. It is used by the grading scripts.
 	 */
 	public Board() {
-		// TODO Implement
+        ships = new ArrayList<Ship>();
+        attacks = new ArrayList<Result>();
 	}
 
 	/*
 	DO NOT change the signature of this method. It is used by the grading scripts.
 	 */
 	public boolean placeShip(Ship ship, int x, char y, boolean isVertical) {
-		// TODO Implement
-		return false;
+        int size = 2;
+        if (ship.getKind().equals("MINESWEEPER")){size = 2;}
+        if (ship.getKind().equals("DESTROYER")){size = 3;}
+        if (ship.getKind().equals("BATTLESHIP")){size = 4;}
+
+        if(isVertical == true){
+            if(x > size){
+                ship.addFeatures(x, y, isVertical, size);
+                ships.add(ship);
+                return true;
+            }
+        }
+        else if(y <= ('J'- (char)size)){
```

Comment:
The math is off here. If the size was 1, y could be equal to j, but this suggests it would have to be less than or equal to 'J' - 1, which is 'I' or anything before.